### ISSUE TITLE
Too high load (Reporter actimeleft)

### ISSUE THREAD

**Issue by juztas** [dmwm/AsyncStageout#4467]
Seems there are multiple tickets got about high load. Checking the htop, most CPU usage comes from voms-proxy-info -actimeleft  which comes from Reporter component. 

As improvement, it could cache the results and reuse them in N minutes. This would speed up Reporter Component and lower the usage of CPU on ASO machine

**Comment by mmascher** [comment]
Has something changed on the machine (new voms-proxy-info version, new ASO that calls this more frequently), or is this simply a result of more load (files to transfer) on the machine?

**Comment by juztas** [comment]
voms-proxy no. ASO is old for long time. Result of bad site. 
https://ggus.eu/?mode=ticket_info&ticket_id=118376

@jmarra13 I suggest to block this site in rest configuration for stageout until site will solve their problems.
As from FTS page, Bad shape of entry means: ` There are submitted but no active, less than 3 active with more than 3 submitted, or a failure rate >= 20%` and for last 7 days its 7.11% success rate. which had almost ~600k FTS documents to transfer.

**Comment by juztas** [comment]
and there are plenty of  new docs submitted. I guess some logic could be added into ASO about that.

**Comment by belforte** [comment]
I was looking at this last night, but the many voms-proxy-info commands seemed harmless, i.e. they take the normal few seconds, use no CPU, and close down quickly. I could not tell what the load is coming from, but I though it should be from running processes (R in top). voms-proxy processes are always listed as sleeping (S), likely waiting on disk read or trying to verify credentials with remote servers.

**Comment by belforte** [comment]
@juztas if one bad site can put so much stress on ASO, I'd say something is quite wrong. I hope we did not design a system which can't cope with one site being down for a week.
Are the 600k documents "all ours" ? i.e. did Crab3 mange to run hundreds of thousands of jobs with stageout sent to Ohio State ?

**Comment by juztas** [comment]
I have to correct myself. there are two diff things, one for reporter component and huge cpu usage which seems to shown mainly voms-proxy and update to coucdb document. This needs to take deeper look.

Result of bad site is another story. Sorry for confusion. So for the last 7 days to Ohio:
https://lcgfts3.gridpp.rl.ac.uk:8449/fts3/ftsmon/#/?vo=&source_se=&dest_se=srm:%2F%2Fcms-0.mps.ohio-state.edu&time_window=168
http://dashb-fts-transfers.cern.ch/ui/#activity=%28ASO%29&date.interval=10080&dst.host=%28cms-0.mps.ohio-state.edu%29&grouping.dst=%28host%29&m.content=%28efficiency,errors,successes%29&p.grouping=server&server=%28bnl,cmsfts3.fnal.gov,fts.hep.pnnl.gov,fts3-pilot.cern.ch,fts3.cern.ch,lcgfts3.gridpp.rl.ac.uk%29
Success rate 7.08% 

For last 24h:
http://dashb-fts-transfers.cern.ch/ui/#activity=%28ASO%29&date.interval=1440&dst.host=%28cms-0.mps.ohio-state.edu%29&grouping.dst=%28host%29&m.content=%28efficiency,errors,successes%29&p.grouping=server&server=%28bnl,cmsfts3.fnal.gov,fts.hep.pnnl.gov,fts3-pilot.cern.ch,fts3.cern.ch,lcgfts3.gridpp.rl.ac.uk%29
Success rate: 17%

And all are ours. From one of the task I see a lot of retries and user resubmitted several times whole task. (Not one task, several tasks)

And error message is recoverable:
TRANSFER  globus_ftp_client: the server responded with an error 421 Service busy: Connection limit exceeded. Please try again later. Closing control connection.

And also he used transfer log files true, which leads to transfer also failed job log files. As it is failing to transfer, ASO is retrying to transfer these files.

**Comment by belforte** [comment]
thanks for these details. About OSU we can't design around the assumption that users are intelligent, careful and react promptly. Clearly as soon as we allow for retries we have to worry about how big a backlog do we allow to be build, where it is stored, and so on. Maybe this part was not thought very carefully in initial design and tests. Blacklisting in server REST config is only a crude palliative.

Indeed load appear due to Reporter processes, it seems there are 300 such processes.

**Comment by belforte** [comment]
looking at kibana, this started on December 23.
![screenshot](https://cloud.githubusercontent.com/assets/1837785/12171109/7cb4b71a-b547-11e5-8164-1e338453de26.png)
very suspicious.
Did some needed component/cron stop running ?

**Comment by juztas** [comment]
strange... from my side nothing changed. Seems this bug is hitting us more now ...

**Comment by belforte** [comment]
something happened yesterday afternoon and then again last nigh, and load is not sort of stable at ~100 instead of ~200. @juztas did you do something ? Or did it happen by itself ?
![screenshot-1](https://cloud.githubusercontent.com/assets/1837785/12195265/23be3550-b5f6-11e5-88b8-b9145f3eee20.png)

by the way this is a matter for e-log more than GIT.

**Comment by juztas** [comment]
yes, I still monitor it and will update ticket. Also will report in the meeting. I am still not sure if this is the case. So more tests are needed

**Comment by HassenRiahi** [comment]
I think the Monitor component got somehow broken. It started somehow processing old transfers and providing their status to the Reporter for the update in the database. Therefore the reporter component got overloaded by the huge update requests (for the current transfers and old ones). 

What I did is restarting the Monitor component aiming that it will be back to the correct behavior. Then if the issue will disappear we need to understand why it happened. 

Pls @juztas could you have a look to the monitoring to check if the CPU usage is back to normal?

**Comment by belforte** [comment]
thanks @HassenRiahi 
things improved around 7:30 pm , but a much bigger step in the right direction happened in early afternoon
![screenshot from 2016-01-08 21 22 08](https://cloud.githubusercontent.com/assets/1837785/12208656/6db2c380-b64e-11eh5-8ef9-5c6c4cad8611.png)
a similar picture for average load, although is was sort of good enough at 7pm that it is difficul to notice the additional downward jump
![screenshot from 2016-01-08 21 24 52](https://cloud.githubusercontent.com/assets/1837785/12208691/985375c6-b64e-11e5-8d88-b36e4521bbd9.png)

**Comment by dciangot** [comment]
This issue may be explained by  #4473 .
If the monitor component have been restarted while the cleanup script didn't work, all the jobs already processed, but not cleaned, would have been taken again from monitor and passed to Reporter.
This could explain all the issues we observed, I think.

**Comment by belforte** [comment]
sounds plausible :-)

**Comment by juztas** [comment]
mm. No. Double checked monitor log which is from 2015-09 and component was not rebooted. So this is not the case.

**Comment by dciangot** [comment]
mmm.. ok..  I'm going to check one more thought, I'm not sure, but even an online change in the monitor.conf could have triggered it

**Comment by dciangot** [comment]
Old misteries remain. I think we may want to close it.

@mmascher @belforte
